### PR TITLE
Release/1.0.0 rc.1

### PR DESCRIPTION
**Pull request by JakubJankowski**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,232 @@
...
+- Enable setting a constant location engine resolution [\#276](https://github.com/ably/ably-asset-tracking-swift/issues/276)
+- Add ability to observe raw locations change in `Subscriber` SDK [\#272](https://github.com/ably/ably-asset-tracking-swift/issues/272)
+- Send batches of location updates [\#265](https://github.com/ably/ably-asset-tracking-swift/issues/265)
+- Upgrade Mapbox dependencies [\#260](https://github.com/ably/ably-asset-tracking-swift/issues/260)
+- Display resolution information in the subscriber example app [\#254](https://github.com/ably/ably-asset-tracking-swift/issues/254)
+- Send resolution to subscriber [\#253](https://github.com/ably/ably-asset-tracking-swift/issues/253)
+- Add switch enabling raw locations publishing in the publisher example app [\#252](https://github.com/ably/ably-asset-tracking-swift/issues/252)
+- Add raw locations to the Subscriber SDK [\#251](https://github.com/ably/ably-asset-tracking-swift/issues/251)
+- Send raw locations from the Publisher [\#250](https://github.com/ably/ably-asset-tracking-swift/issues/250)
+- Route is not recalculated when user is off route [\#184](https://github.com/ably/ably-asset-tracking-swift/issues/184)
```
Comment (KacperKluka) [inline CHANGELOG.md:17]: This one was just closed (without implementation) so it shouldn't be in the changelog :wink:

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,232 @@
...
+- Implement `RSC7d` \(Ably-Agent header\) [\#172](https://github.com/ably/ably-asset-tracking-swift/issues/172)
+- Implement stopping strategy [\#165](https://github.com/ably/ably-asset-tracking-swift/issues/165)
+- Test issue ignore [\#163](https://github.com/ably/ably-asset-tracking-swift/issues/163)
```
Comment (KacperKluka) [inline CHANGELOG.md:20]: This probably shouldn't be here :wink:

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,232 @@
...
+- Send skipped location updates [\#135](https://github.com/ably/ably-asset-tracking-swift/issues/135)
+- Add token based authentication [\#134](https://github.com/ably/ably-asset-tracking-swift/issues/134)
+- Modify Publisher example app to work in background [\#110](https://github.com/ably/ably-asset-tracking-swift/issues/110)
+- Enable setting a constant location engine resolution [\#277](https://github.com/ably/ably-asset-tracking-swift/pull/277) ([lukasz-szyszkowski](https://github.com/lukasz-szyszkowski))
```
Comment (KacperKluka) [inline CHANGELOG.md:24]: Since this is a PR shouldn't it be under `Merged pull requests`? :thinking:

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,232 @@
...
+- Release to Swift Package Manager \(SPM\) [\#148](https://github.com/ably/ably-asset-tracking-swift/issues/148)
+- Change names of client type values from presence data [\#136](https://github.com/ably/ably-asset-tracking-swift/issues/136)
+- EnhancedLocationUpdateMessage should be sent and received as a single message, not an array. [\#126](https://github.com/ably/ably-asset-tracking-swift/issues/126)
+- Use Ably and channel state when determining the trackable state in the Subscriber and Publisher SDK [\#120](https://github.com/ably/ably-asset-tracking-swift/issues/120)
+- The trackable state should be updated in the Publisher only after the state is different from the previous one. [\#115](https://github.com/ably/ably-asset-tracking-swift/issues/115)
+- Allow to call Publisher track\(\) method multiple times [\#112](https://github.com/ably/ably-asset-tracking-swift/issues/112)
+- Handle stopped state of the Subscriber [\#109](https://github.com/ably/ably-asset-tracking-swift/issues/109)
+- Fix lint errors for podspec [\#41](https://github.com/ably/ably-asset-tracking-swift/issues/41)
```
Comment (KacperKluka) [inline CHANGELOG.md:41]: This has also been just closed without fixing, so let's remove it from the list :wink:

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,232 @@
...
+- Make `ConnectionConfiguration` initialiser visible in Objective-C  [\#192](https://github.com/ably/ably-asset-tracking-swift/issues/192)
+- Tests refactoring [\#185](https://github.com/ably/ably-asset-tracking-swift/issues/185)
+- Handle exception thrown when sending enhanced location update fails [\#178](https://github.com/ably/ably-asset-tracking-swift/issues/178)
+-  Create code snippets for landing page [\#173](https://github.com/ably/ably-asset-tracking-swift/issues/173)
+- Conform ReadMe and create Contributing Document [\#171](https://github.com/ably/ably-asset-tracking-swift/issues/171)
+- Share build outputs instead of building the library 3 times in separate workflows [\#168](https://github.com/ably/ably-asset-tracking-swift/issues/168)
+- Add Swift Package to Swift Package Index [\#167](https://github.com/ably/ably-asset-tracking-swift/issues/167)
+- iOS Integration Testing set up [\#166](https://github.com/ably/ably-asset-tracking-swift/issues/166)
+- Final test sync after repo has been renamed \(ignore\) [\#164](https://github.com/ably/ably-asset-tracking-swift/issues/164)
+- Ignore this issue [\#162](https://github.com/ably/ably-asset-tracking-swift/issues/162)
```
Comment (KacperKluka) [inline CHANGELOG.md:77]: Those two should probably also be removed :sweat_smile:

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,232 @@
...
+- Supporting location button \(in SDK and Example apps\) [\#160](https://github.com/ably/ably-asset-tracking-swift/issues/160)
+- Create Swift Package Collection to conveniently store all Ably SDKs in one group [\#159](https://github.com/ably/ably-asset-tracking-swift/issues/159)
```
Comment (KacperKluka) [inline CHANGELOG.md:79]: This one is also not implemented so let's remove it :wink:

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,232 @@
...
+- Send notification about trip started on a meta channel \(iOS\) [\#158](https://github.com/ably/ably-asset-tracking-swift/issues/158)
+- Error `nw_protocol_get_quic_image_block_invoke dlopen libquic failed` when running app [\#157](https://github.com/ably/ably-asset-tracking-swift/issues/157)
+- Create Cocoapods installation instructions in Readme for SDK users [\#155](https://github.com/ably/ably-asset-tracking-swift/issues/155)
+- Consider merging both SDKs into one SDK, or into one Xcode project [\#154](https://github.com/ably/ably-asset-tracking-swift/issues/154)
```
Comment (KacperKluka) [inline CHANGELOG.md:83]: This also seems to be just closed - let's remove it :thinking:

**Review by KacperKluka — CHANGES_REQUESTED** [review]
That's a really long list and I've tried to find as many issues as I could but I've might missed some of them :thinking: If you could double-check them and look for "not implemented but closed" issues as well it would be great :grin:

File: CHANGELOG.md (hunk 3 above)
Comment (JakubJankowski) [reply CHANGELOG.md:24]: Yeah, it is duplicated in a couple of places. I'll leave it only in the `Merged pull requests` section.

**Comment by JakubJankowski** [comment]
Yeaah, the doc generating tool isn't very smart, should've anticipated that. I'll look through all the issues again as well and clean it up a bit.

**Comment by JakubJankowski** [comment]
In [79fe933](https://github.com/ably/ably-asset-tracking-swift/pull/322/commits/79fe933091e23dfaba40d17a95f20933f6f818e0) I've cleaned up some changelog entries that seemed useless from a point of view of a developer using the sdk.

There's still a lot of entries though I'm afraid, so any suggestions on how to improve that are welcome.

I've readded the `Enable setting a constant location engine resolution` entry the enhancements section, since it seemed like an actually important enhancement, even if it's also mentioned in the `Merged Pull Requests` section.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,200 @@
 # Changelog
 
+## [1.0.0-Release Candidate 1](https://github.com/ably/ably-asset-tracking-swift/tree/v1.0.0-rc.1)
+[Full Changelog](https://github.com/ably/ably-asset-tracking-swift/compare/v1.0.0-beta.1...v1.0.0-rc.1)
```
Comment (QuintinWillison) [inline CHANGELOG.md:4]: To stay consistent with previous changelog entries, please insert a line of vertical whitespace above this line, separating it from the title. Some readers may consume this document in raw format, not rendered.

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,200 @@
...
+- Check: Zooming does not improve resolution: Resolution-type events does not get sent [\#255](https://github.com/ably/ably-asset-tracking-swift/issues/255)
+- Update metadata format pushed with location update [\#246](https://github.com/ably/ably-asset-tracking-swift/issues/246)
+- Creating of `SwiftUI` example app [\#231](https://github.com/ably/ably-asset-tracking-swift/issues/231)
+- Update CONTRIBUTING.md about version bumping [\#223](https://github.com/ably/ably-asset-tracking-swift/issues/223)
+- Change `Codable` classes to structs [\#222](https://github.com/ably/ably-asset-tracking-swift/issues/222)
+- Remove Objective-C compatibility [\#218](https://github.com/ably/ably-asset-tracking-swift/issues/218)
+- We need to check how Swift-idiomatic the library is [\#217](https://github.com/ably/ably-asset-tracking-swift/issues/217)
+- Remove unused libs [\#211](https://github.com/ably/ably-asset-tracking-swift/issues/211)
+- Move `.xcworkspace` to `Examples` directory [\#203](https://github.com/ably/ably-asset-tracking-swift/issues/203)
```
Comment (QuintinWillison) [inline CHANGELOG.md:56]: Doesn't represent a change that carries any value for the SDK user / app developer, so I would remove this line.

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,200 @@
...
+- Clear and visible UI indications onSuccess and onFailure [\#76](https://github.com/ably/ably-asset-tracking-swift/issues/76)
+- Create a project with snippets of Objective-C code calling our SDKs [\#74](https://github.com/ably/ably-asset-tracking-swift/issues/74)
+- iOS: Remove sending raw updates from the publisher SDK [\#73](https://github.com/ably/ably-asset-tracking-swift/issues/73)
+- Use AssetTrackingError where possible [\#66](https://github.com/ably/ably-asset-tracking-swift/issues/66)
+- Create ARTRealtime instances with `clientId` parameter [\#65](https://github.com/ably/ably-asset-tracking-swift/issues/65)
+- Adopt callback interface changes from Android - single handler, not discrete onSuccess and onError [\#64](https://github.com/ably/ably-asset-tracking-swift/issues/64)
+- Accept new `LocationUpdate` format on the Subscriber side [\#63](https://github.com/ably/ably-asset-tracking-swift/issues/63)
+- Handle `Publisher.stop()` correctly [\#51](https://github.com/ably/ably-asset-tracking-swift/issues/51)
+- Find a better way of importing sources from Core to Publisher and Subscriber [\#43](https://github.com/ably/ably-asset-tracking-swift/issues/43)
```
Comment (QuintinWillison) [inline CHANGELOG.md:99]: This feels like something that does not carry any value for the SDK user / app developer. I would remove it.

**Review by QuintinWillison — CHANGES_REQUESTED** [review]
Thanks for this. It looks like there are definitely more line items that can be removed (I've commented on a couple but will leave it to you to find others). If something relates to tests or internal project structure then it definitely should not be here. A user is coming to this document in order to work out the changes we've made that will or could have material impact on them in respect of their use of this SDK in their app.

File: CHANGELOG.md (hunk 8 above)
Comment (JakubJankowski) [reply CHANGELOG.md:5]: Fixed in [e2cbd82](https://github.com/ably/ably-asset-tracking-swift/pull/322/commits/e2cbd82ad81e7780ba1e00d6cf4f5e581768340d)

File: CHANGELOG.md (hunk 9 above)
Comment (JakubJankowski) [reply CHANGELOG.md]: Removed in [e2cbd82](https://github.com/ably/ably-asset-tracking-swift/pull/322/commits/e2cbd82ad81e7780ba1e00d6cf4f5e581768340d)

File: CHANGELOG.md (hunk 10 above)
Comment (JakubJankowski) [reply CHANGELOG.md]: Removed in [e2cbd82](https://github.com/ably/ably-asset-tracking-swift/pull/322/commits/e2cbd82ad81e7780ba1e00d6cf4f5e581768340d). I also made another pass over this file and removed some additional entries.

**Review by QuintinWillison — APPROVED** [review]
Thanks for making all those tweaks. Looking great now, thanks. 💪

**Comment by JakubJankowski** [comment]
Oh, looks like I'm not authorized to merge pull requests in this repo 😅 Would you mind merging it for me (or bumping my permissions)? @QuintinWillison

**Pull request merged**